backend/src/main/java/com/htecgroup/skynest/service/impl/FolderServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package com.htecgroup.skynest.service.impl;
+
+import com.htecgroup.skynest.annotation.ParentFolderIsInTheSameBucket;
+import com.htecgroup.skynest.model.dto.LoggedUserDto;
+import com.htecgroup.skynest.model.entity.BucketEntity;
+import com.htecgroup.skynest.model.entity.FolderEntity;
+import com.htecgroup.skynest.model.entity.UserEntity;
+import com.htecgroup.skynest.model.request.FolderCreateRequest;
+import com.htecgroup.skynest.model.response.FolderResponse;
+import com.htecgroup.skynest.repository.FolderRepository;
+import com.htecgroup.skynest.repository.UserRepository;
+import com.htecgroup.skynest.service.BucketService;
+import com.htecgroup.skynest.service.CurrentUserService;
+import com.htecgroup.skynest.service.FolderService;
+import lombok.AllArgsConstructor;
+import org.modelmapper.ModelMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+import javax.validation.Valid;
+
+@Service
+@AllArgsConstructor
+@Validated
+public class FolderServiceImpl implements FolderService {
+
+  private ModelMapper modelMapper;
+  private FolderRepository folderRepository;
+  private CurrentUserService currentUserService;
+
+  private UserRepository userRepository;
+
+  private BucketService bucketService;
+
+  @Override
+  public FolderResponse createFolder(
```

Comment:
According to me this method needs complete rewriting. It looks too complicated with a lot of things tightly coupled. 
If I understand correctly FolderEntity has a FK to Bucket and User, that means that you need only the ids not the whole objects, since you are not creating neither the buckets nor the user, you just need the reference.

Having that said, 
```
BucketEntity bucketEntity =
        modelMapper.map(
            bucketService.findBucketById(folderCreateRequest.getBucketId()), BucketEntity.class);
```
and 
`UserEntity currentUser = userRepository.getById(loggedUserDto.getUuid());`
are totally unnecessary.

Probably same can be done with `parentFolder` and use just the id, if you already check it in the annotation it self